testsuite/forge/src/backend/k8s/fullnode.rs

```diff
@@ -0,0 +1,738 @@
+        }),
+        ..Service::default()
+    })
+}
+
+/// Create a fullnode StatefulSet given some templates from the validator
+fn create_fullnode_stateful_set(
+    fullnode_name: String,
+    fullnode_image: String,
+    fullnode_genesis_secret_name: String,
+    fullnode_node_config_config_map_name: String,
+    validator_stateful_set: StatefulSet,
+    validator_data_volume: PersistentVolumeClaim,
+) -> Result<StatefulSet> {
+    let labels_map: BTreeMap<String, String> = create_fullnode_labels(fullnode_name.clone());
+
+    // create the fullnode data volume
+    let data_volume = create_fullnode_persistent_volume_claim(validator_data_volume)?;
+
+    // build the fullnode stateful set
+    let mut fullnode_stateful_set = StatefulSet::default();
+    fullnode_stateful_set.metadata.name = Some(fullnode_name.clone());
+    fullnode_stateful_set.metadata.labels = Some(labels_map.clone());
+    fullnode_stateful_set.spec = Some(StatefulSetSpec {
+        service_name: fullnode_name,
+        selector: LabelSelector {
+            match_labels: Some(labels_map.clone()),
+            ..LabelSelector::default()
+        },
+        volume_claim_templates: Some(vec![data_volume]), // a PVC that is created directly by the StatefulSet, and owned by it
+        template: PodTemplateSpec {
+            metadata: Some(ObjectMeta {
+                labels: Some(labels_map),
+                ..ObjectMeta::default()
+            }),
+            spec: Some(PodSpec {
+                containers: vec![Container {
+                    image: Some(fullnode_image),
+                    command: Some(vec![
+                        "/usr/local/bin/aptos-node".to_string(),
+                        "-f".to_string(),
+                        format!("/opt/aptos/etc/{}", FULLNODE_CONFIG_MAP_KEY),
+                    ]),
+                    volume_mounts: Some(vec![
+                        VolumeMount {
+                            mount_path: APTOS_CONFIG_VOLUME_PATH.to_string(),
+                            name: APTOS_CONFIG_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                        VolumeMount {
+                            mount_path: APTOS_DATA_VOLUME_PATH.to_string(),
+                            name: APTOS_DATA_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                        VolumeMount {
+                            mount_path: GENESIS_CONFIG_VOLUME_PATH.to_string(),
+                            name: GENESIS_CONFIG_VOLUME_NAME.to_string(),
+                            ..VolumeMount::default()
+                        },
+                    ]),
+                    // specifically, inherit resources, env,ports, securityContext from the validator
+                    ..validator_stateful_set
+                        .spec
+                        .as_ref()
+                        .unwrap()
```

Comment:
I dont think we should unwrap here, we can raise the error upwards, but unwrapping will panic in a cryptic way